src/analysis/normed_space/box_subadditive.lean

```diff
@@ -0,0 +1,541 @@
+  rw seq_succ,
+  exact (next _ _ _).coe_prop.2.1
+end
+
+lemma tendsto_dist_seq_left_right (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) :
+  tendsto (λ n, dist (seq hf hg I n).left (seq hf hg I n).right) at_top (𝓝 0) :=
+begin
+  simp only [dist_seq_left_right],
+  exact tendsto_const_nhds.div_at_top (tendsto_pow_at_top_at_top_of_one_lt one_lt_two)
+end
+
+lemma fix_mem_Inter_seq (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c):
+  fix hf hg I ∈ ⋂ n, Icc (seq hf hg I n).left (seq hf hg I n).right :=
+csupr_mem_Inter_Icc_of_mono_incr_of_mono_decr (seq_left_mono hf hg I) (seq_right_mono hf hg I) $
+  λ n, (seq hf hg I n).le
+
+lemma fix_mem_seq (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) (n : ℕ) :
+  fix hf hg I ∈ Icc (seq hf hg I n).left (seq hf hg I n).right :=
+have _ := set.mem_Inter.1 (fix_mem_Inter_seq hf hg I) n, this
+
+lemma fix_mem (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) :
+  fix hf hg I ∈ Icc I.left I.right :=
+fix_mem_seq hf hg I 0
+
+lemma fix_mem_set (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) :
+  fix hf hg I ∈ s :=
+I.sub $ fix_mem hf hg I
+
+lemma tendsto_left_nhds_fix (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) :
+  tendsto (λ n, (seq hf hg I n).left) at_top (𝓝 (fix hf hg I)) :=
+begin
+  refine (tendsto_iff_dist_tendsto_zero.2 $
+    squeeze_zero (λ _, dist_nonneg) (λ n, _) (tendsto_dist_seq_left_right hf hg I)),
+  refine (dist_pi_le_iff dist_nonneg).2 (λ i, le_trans _ (dist_le_pi_dist _ _ i)),
+  exact real.dist_left_le_of_mem_interval (set.Icc_subset_interval $
+    ⟨(fix_mem_seq hf hg I _).1 _, (fix_mem_seq hf hg I _).2 _⟩)
+end
+
+lemma tendsto_left_nhds_within_fix (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) :
+  tendsto (λ n, (seq hf hg I n).left) at_top (𝓝[Icc I.left (fix hf hg I)] (fix hf hg I)) :=
+tendsto_inf.2 ⟨tendsto_left_nhds_fix hf hg I, tendsto_principal.2 $ eventually_of_forall $
+  λ n, ⟨seq_left_mono hf hg I (zero_le n), (fix_mem_seq hf hg I n).1⟩⟩
+
+lemma tendsto_right_nhds_fix (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) :
+  tendsto (λ n, (seq hf hg I n).right) at_top (𝓝 (fix hf hg I)) :=
+(tendsto_left_nhds_fix hf hg I).congr_dist (tendsto_dist_seq_left_right hf hg I)
+
+lemma tendsto_right_nhds_within_fix (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) :
+  tendsto (λ n, (seq hf hg I n).right) at_top (𝓝[Icc (fix hf hg I) I.right] (fix hf hg I)) :=
+tendsto_inf.2 ⟨tendsto_right_nhds_fix hf hg I, tendsto_principal.2 $ eventually_of_forall $
+  λ n, ⟨(fix_mem_seq hf hg I n).2, seq_right_mono hf hg I (zero_le n)⟩⟩
+
+lemma le_mul_of_forall_eventually_le_mul (hle : l ≤ u)
```

Comment:
docstring? (This is really more general than the next one, so it deserves a docstring IMHO)